### ISSUE TITLE
Patterns for single products > Ensure the user can select the product they wish to use the pattern with without affecting the preview 

### ISSUE THREAD

**Issue by nefeline** [woothemes/woocommerce#42324]
This discussion started on https://github.com/woocommerce/woocommerce-blocks/pull/9935/files#r1243957131 : 

Currently, as @albarin pointed out, when relying on the Single Product block for rendering the pattern, we can't display the following product selection list without affecting the preview for the Pattern:

<img width="661" alt="Screenshot 2023-06-28 at 13 08 27" src="https://github.com/woocommerce/woocommerce-blocks/assets/15730971/c697a02b-f1dc-4e31-a084-fa29e207aac8">

<img width="367" alt="249402546-ea4409b9-ac02-46ea-9279-74c3a22893ba" src="https://github.com/woocommerce/woocommerce-blocks/assets/15730971/9c16eb47-f398-461a-ba82-2a8f68d86419">

Ideally, we should be able to:

- Allow users to select the product they want to use the pattern with, and while doing so,
- Allow them to preview the pattern design without any problems: the preview should match the designed pattern rather than the list of products.

This is an issue for us to investigate what can be done to achieve this (most likely, changes on Gutenberg are required to achieve this goal).

**Comment by danielwrobert** [comment]
If we’re blocked on the preview issue noted above, perhaps another approach for improvement that we could look into would be to allow the user to change the product _after_ insertion.

Here there would be a default product for insertion and and then the merchant/user would be able to change the product from there.

IMO, this capability would also be useful _alongside_ the ability to select upon insertion, so it’s something that still adds value even if we can resolve the preview issue later on.

**Comment by albarin** [comment]
> 

@nefeline @danielwrobert Just want to clarify, it is actually possible to change the product after insertion since the pattern is only a `Single Product` block with some custom styles 👇 

https://github.com/woocommerce/woocommerce-blocks/assets/186112/9a4430c2-ab44-4dc2-a0d6-969026cc4769

It would be nice to have a preview with a random product and then, after insertion, have the product selector open right away, but I think what we currently have is a good solution for now.

**Comment by nefeline** [comment]
> Just want to clarify, it is actually possible to change the product after insertion since the pattern is only a Single Product block with some custom styles

Nice! TIL

> It would be nice to have a preview with a random product and then, after insertion, have the product selector open right away, but I think what we currently have is a good solution for now.

Considering that we already have a way for users to switch the product displayed while preserving the styles for the pattern, I agree that what we have now is good enough at this first moment.

**Comment by danielwrobert** [comment]
@albarin thanks for the clarification, makes sense!

It sounds like we already have what I was suggesting then and I agree with you and @nefeline that this is a good solution for now.

**Comment by kkoteen** [comment]
Closing as the core need is resolved in the above comments.
